fluentd/configs.d/user/fluent.conf

```diff
@@ -23,6 +23,12 @@
   @include configs.d/openshift/filter-k8s-record-transform.conf
   @include configs.d/openshift/filter-syslog-record-transform.conf
   @include configs.d/openshift/filter-viaq-data-model.conf
+  # generate uuids at source mitigate creating duplicate records
+  # requires fluent-plugin-elasticsearch
+  <filter **>
+    @type elasticsearch_genid
+    hash_id_key _openshift_es_genid
```

Comment:
is there a reason we are using this particular key only to map and remove it later?

Reply:
This is what is recommended by the filter.  The filter generates the key, copies the value to the '_id' field, and then comments that ES does not like records with '_' so it is removed prior to submission

Reply:
I don't think this file is used.
The "real" fluent.conf comes from ansible.
If we want this in the product by default, we'll need to create configs.d/openshift/filter-post-esid.conf or something like that with this config.  And - if we want to easily disable it, we'll need to make the inclusion of this file conditional based on an env var.

Reply:
ack, i read the documentation for the filter type. I didn't realize that `id_key` in the output plugin denotes what should be used as that key... I read it as it was setting another key to be that value.